pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -122,33 +123,52 @@ func (s *FlinkStateMachine) shouldRollback(ctx context.Context, application *v1b
 }
 
 func (s *FlinkStateMachine) Handle(ctx context.Context, application *v1beta1.FlinkApplication) error {
-	currentPhase := application.Status.Phase
-	if _, ok := s.metrics.stateMachineHandlePhaseMap[currentPhase]; !ok {
-		errMsg := fmt.Sprintf("Invalid state %s for the application", currentPhase)
-		return errors.New(errMsg)
-	}
-	timer := s.metrics.stateMachineHandlePhaseMap[currentPhase].Start(ctx)
-	successTimer := s.metrics.stateMachineHandleSuccessPhaseMap[currentPhase].Start(ctx)
-
-	defer timer.Stop()
-	updateStatus, err := s.handle(ctx, application)
 
-	// Update k8s object
-	if updateStatus {
-		now := v1.NewTime(s.clock.Now())
-		application.Status.LastUpdatedAt = &now
-		updateAppErr := s.k8Cluster.UpdateStatus(ctx, application)
-		if updateAppErr != nil {
+	for {
```

Comment:
Infinite loop looks strange here and can be rewritten by using channels.

Reply:
> can be rewritten by using channels.
I have no idea how to do that.

Reply:
@mdreizin seems like channels might be overkill in this situation.  I agree the infinite loop isn't very 'go-ish' but since we aren't really looking for concurrency it should be a problem.  thoughts?

Reply:
@rtluckie it looks like @mdreizin agrees.